pandas/tests/io/test_excel.py

```diff
@@ -163,20 +164,96 @@ def test_usecols_str(self, ext):
-                               usecols='A,C,D')
+                               usecols_excel='A,C,D')
         df3 = self.get_exceldf('test1', ext, 'Sheet2', skiprows=[1],
-                               index_col=0, usecols='A,C,D')
+                               index_col=0, usecols_excel='A,C,D')
         # TODO add index to xls file
         tm.assert_frame_equal(df2, df1, check_names=False)
         tm.assert_frame_equal(df3, df1, check_names=False)
 
         df1 = dfref.reindex(columns=['B', 'C'])
         df2 = self.get_exceldf('test1', ext, 'Sheet1', index_col=0,
-                               usecols='A,C:D')
+                               usecols_excel='A,C:D')
+        df3 = self.get_exceldf('test1', ext, 'Sheet2', skiprows=[1],
+                               index_col=0, usecols_excel='A,C:D')
+        tm.assert_frame_equal(df2, df1, check_names=False)
+        tm.assert_frame_equal(df3, df1, check_names=False)
+
+    @pytest.mark.parametrize("columns,usecols_excel,parse_cols", [
+        (['A', 'B', 'C'], ['A:D'], ['A:D']),
+        (['B', 'C'], ['A', 'C', 'D'], ['A', 'C', 'D']),
+        (['B', 'C'], ['A', 'C:D'], ['A', 'C:D'])
+    ])
+    # GH18273 - read_excel return empty dataframe when using usecols_excel
+    # as a list of strings
+    def test_usecols_excel_str_list(self, ext, columns, usecols_excel,
+                                    parse_cols):
+
+        dfref = self.get_csv_refdf('test1')
+
+        df1 = dfref.reindex(columns=columns)
+        df2 = self.get_exceldf('test1', ext, 'Sheet1', index_col=0,
+                               usecols_excel=usecols_excel)
         df3 = self.get_exceldf('test1', ext, 'Sheet2', skiprows=[1],
-                               index_col=0, usecols='A,C:D')
+                               index_col=0, usecols_excel=usecols_excel)
+
+        with tm.assert_produces_warning(FutureWarning):
+            df4 = self.get_exceldf('test1', ext, 'Sheet2', skiprows=[1],
+                                   index_col=0, parse_cols=parse_cols)
+
+        # TODO add index to xls, read xls ignores index name ?
         tm.assert_frame_equal(df2, df1, check_names=False)
         tm.assert_frame_equal(df3, df1, check_names=False)
+        tm.assert_frame_equal(df4, df1, check_names=False)
+
+    def test_usecols_diff_positional_int_columns_order(self, ext):
+
+        dfref = self.get_csv_refdf('test1')
+
+        df1 = dfref.reindex(columns=['A', 'C'])
+        df2 = self.get_exceldf('test1', ext, 'Sheet1', usecols=[0, 2])
+        df3 = self.get_exceldf('test1', ext, 'Sheet1', usecols=[2, 0])
+
+        tm.assert_frame_equal(df2, df1, check_names=False)
+        tm.assert_frame_equal(df3, df2, check_names=False)
+
+    def test_usecols_diff_positional_str_columns_order(self, ext):
+
+        df1 = self.get_csv_refdf('test1')[['B', 'D']]
+
+        with tm.assert_produces_warning(UserWarning):
+            df2 = self.get_exceldf('test1', ext, 'Sheet1', usecols=['B', 'D'])
```

Comment:
As state above I'm not sure this should warn, since it didn't previously work.  But I would like a test for the warning, something like `usecols='B,D'`

Reply:
I can't pass a string to usecols because if I do it throws an error. I already tested it.
So, does not make sense maintain the warning just to a string type parameter unless you do not want the error be thrown.

Do I should remove the warning, then?

The other changes you asked me are done. I'm just waiting to know what to do about this warning to commit again.